### DISCUSSION TITLE
Can you use targets over multiple EC2 instances?

### DISCUSSION THREAD

**Discussion by adrianadepalma** [ropensci/targets#1068] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I'm a bit of a novice with the targets package and with AWS. I was just getting the hang of targets in a local workflow, but we're now considering moving to EC2. In order to balance cost and speed, some targets will loop over inputs across multiple EC2 instances; some targets will use a single EC2 instance but the specs of the instance will vary between targets. Is it at all possible to use targets in a workflow like this? Thanks in advance for any help.

**Comment by wlandau** [comment, marked as answer]
I hope it will be in the future, it's definitely one of my goals for integration with [`crew`](https://wlandau.github.io/crew): https://books.ropensci.org/targets/crew.html. Right now though, `crew` only has plugins for local workers and for [traditional computing clusters](https://wlandau.github.io/crew.cluster).

When it does become available, regardless of who writes it (could be me), an EC2 [launcher plugin](https://wlandau.github.io/crew/articles/launcher_plugins.html) will rely on direct TCP connections from the local R session with those EC2s. In most cases, I would recommend starting that local R session on AWS (e.g. through an AWS-backed interactive RStudio environment) on the same local network (security group, VPC) that the EC2s will join. Like many similar HPC tools, `crew` assumes that the workers and controlling process share a local network. (Currently these network connections are unencrypted, but there are plans for TLS via `nanonext` and `mirai`.)
